src/global-messages/GlobalMessage.tsx

```diff
@@ -24,34 +30,30 @@ const GlobalMessage = ({globalMessageContext, style}: Props) => {
     addDismissedGlobalMessages,
   } = useGlobalMessagesState();
 
-  const globalMessages = globalMessageContext
+  const globalMessages = showAllMessages
+    ? findGlobalMessages()
+    : globalMessageContext
     ? findGlobalMessages(globalMessageContext)
-    : undefined;
+    : [];
 
-  if (!globalMessages?.length) {
+  if (globalMessages.length === 0 && !showAllMessages) {
     return null;
   }
```

Comment:
It was like this before, but I think we can move this check which returns `null` to after the two filters checking if dismissed and checking the time range.